server/server.go

```diff
@@ -135,10 +135,14 @@ func newServer(logger log.Logger, upstream, directory string, cacheEnabled bool)
 		ObjectBatchActionURLRewriter: DefaultObjectBatchActionURLRewriter,
 	}
 
-	_, err = rand.Read(s.hmacKey[:])
-	if err != nil {
-		return nil, err
-	}
+        if hmacKey != nil {
+		copy(s.hmacKey[:], hmacKey)
+        } else {
+		_, err = rand.Read(s.hmacKey[:])
+		if err != nil {
+			return nil, err
+		}
+        }
```

Comment:
@aaronborden-rivian Can you run `go fmt` for this file?

`go fmt` automatically formats Go files to ensure that they're always consistent. I probably should add a CI linter job to detect this.